x-pack/legacy/plugins/siem/public/components/fields_browser/header.tsx

```diff
@@ -96,27 +99,44 @@ const TitleRow = React.memo<{
   isEventViewer?: boolean;
   onOutsideClick: () => void;
   onUpdateColumns: OnUpdateColumns;
-}>(({ isEventViewer, onOutsideClick, onUpdateColumns }) => (
-  <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" direction="row" gutterSize="none">
-    <EuiFlexItem grow={false}>
-      <EuiTitle data-test-subj="field-browser-title" size="s">
-        <h2>{i18n.CUSTOMIZE_COLUMNS}</h2>
-      </EuiTitle>
-    </EuiFlexItem>
-
-    <EuiFlexItem grow={false}>
-      <EuiButtonEmpty
-        data-test-subj="reset-fields"
-        onClick={() => {
-          onUpdateColumns(isEventViewer ? eventsDefaultHeaders : defaultHeaders);
-          onOutsideClick();
-        }}
-      >
-        {i18n.RESET_FIELDS}
-      </EuiButtonEmpty>
-    </EuiFlexItem>
-  </EuiFlexGroup>
-));
+}>(({ isEventViewer, onOutsideClick, onUpdateColumns }) => {
+  const timelineTypeContext = useTimelineTypeContext();
+  const handleResetColumns = useCallback(() => {
+    let resetDefaultHeaders = defaultHeaders;
+    if (isEventViewer) {
+      if (timelineTypeContext.documentType?.toLocaleLowerCase() === 'alerts') {
+        resetDefaultHeaders = alertsHeaders;
+      } else if (timelineTypeContext.documentType?.toLocaleLowerCase() === 'signals') {
```

Comment:
nit: May want to compare against `i18n.SIGNALS_DOCUMENT_TYPE` in case that changes at any point. Ideally we'd have a `type` prop similar as to what we do elsewhere, but we can revisit that when The Great Timeline Refactor occurs.

Reply:
that's fixing the bug, but I will re-iterate with another PR and add `type DocumentType = 'alert' | 'signals' | 'events;'`